hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkTable.java

```diff
@@ -53,7 +53,7 @@ protected HoodieSparkTable(HoodieWriteConfig config, HoodieEngineContext context
   public static <T> HoodieSparkTable<T> create(HoodieWriteConfig config, HoodieEngineContext context) {
     HoodieTableMetaClient metaClient =
         HoodieTableMetaClient.builder().setConf(context.getHadoopConf().get()).setBasePath(config.getBasePath())
-            .setLoadActiveTimelineOnLoad(true).setConsistencyGuardConfig(config.getConsistencyGuardConfig())
```

Comment:
Note, I even think we don't need this parameter `loadActiveTimelineOnLoad`,  we can always lazy get it

Reply:
For most of the cases, the timeline load is necessary.

Reply:
Ok, revert it in this PR, will consider more